src/main/java/com/faforever/client/mod/ModManagerController.java

```diff
@@ -103,6 +116,26 @@ public void initialize() {
     setCloseable(true);
   }
 
+  private void onTextChange(String newValue){
+    if(newValue.isEmpty()){
+      modVersionFilteredList.setPredicate(viewToggleGroup.getSelectedToggle() == uiModsButton ? UI_FILTER : SIM_FILTER);
+    } else {
+      modVersionFilteredList.setPredicate(getCombinedFilter(newValue));
+    }
```

Comment:
The predicate can be set just once and doesn't need the if statement here since you can check for empty in the predicate itself